### PR TITLE
#1944 Only show RAs virtual qf and sf scores after score display toggle is set

### PR DESCRIPTION
**Pull request by stenington**
I think this will prevent RAs from seeing quarterfinals virtual scores and any semifinals scores until scores are displayed by setting the admin setting.

### CODE REVIEW

File: app/views/regional_ambassador/score_details/show.en.html.erb [hunk 1]
```diff
@@ -8,9 +8,11 @@
     <dt>Event:</dt>
     <dd><%= @submission.team.event.name %></dd>
 
-    <% if @submission.semifinalist? %>
-      <dt>Semifinals average:</dt>
-      <dd><%= @submission.semifinals_average_score %></dd>
+    <% if current_account.is_admin? or SeasonToggles.display_scores? %>
```
Comment (stenington) [inline app/views/regional_ambassador/score_details/show.en.html.erb]: Admins and RAs see the same view, so I just check if the user is an admin here. I'm not sure if it's better to split out the two templates though, I haven't seen a lot of other screens that share the template between the two.

File: app/controllers/regional_ambassador/scores_controller.rb [hunk 2]
```diff
@@ -44,7 +47,7 @@ def grid_params
           end
         ),
         current_account: current_account,
-        round: params.fetch(:round) { round },
+        round: round,
```
Comment (stenington) [inline app/controllers/regional_ambassador/scores_controller.rb]: There was actually a bug for RAs where the round parameter was never carried through no matter how they set it. It always defaulted to the current round.

This solution presents them with both quarterfinals and semifinals as options, but ignores their choice if scores aren't set to display in the season toggles. It's a crappy user experience, but slightly better than the broken experience they would have had before. That said, I can look into maybe not even showing the options that they won't be able to pick if you think it's worth it.

**Comment by stenington** [comment]
I need to fix tests and try to write some new ones, maybe.

**Comment by stenington** [comment]
@adrichey This is ready for review, I think. There are a couple places where I took the easy way out to see if it would pass code review, but I'm fine with you calling me on that. 😁

File: app/views/regional_ambassador/score_details/show.en.html.erb (hunk 1 above)
Comment (adrichey) [reply app/views/regional_ambassador/score_details/show.en.html.erb]: I am okay with this. I think the admin check is fine. I think the alternative doesn't really give us much here.

File: app/controllers/regional_ambassador/scores_controller.rb (hunk 2 above)
Comment (adrichey) [reply app/controllers/regional_ambassador/scores_controller.rb]: For now, I think this is good. Thanks for catching this.

**Pull request merged**